Source/JavaScriptCore/dfg/DFGArrayMode.cpp

```diff
@@ -661,6 +661,9 @@ bool ArrayMode::alreadyChecked(Graph& graph, Node* node, const AbstractValue& va
     case Array::AnyTypedArray:
         return speculationChecked(value.m_type, SpecTypedArrayView);
 
+    case Array::DataView:
+        return speculationChecked(value.m_type, SpecDataViewObject);
```

Comment:
Since `DataView` is not array, I don't think we should extend ArrayMode. Just use DataViewUse speculation and that's fine.